blaze/compute/tests/test_postgresql_compute.py

```diff
@@ -410,6 +408,14 @@ def test_distinct_on(sql):
     assert odo(computation, tuple) == (('a', 1), ('b', 2))
 
 
+def test_relabel_columns_over_selection(big_sql):
+    t = symbol('t', discover(big_sql))
+    result = compute(t[t['B'] == 2].relabel(B='b'),
+                     big_sql, return_type=pd.DataFrame)
+    expected = pd.DataFrame({"A": ['a'], "b": [2]})
+    tm.assert_frame_equal(result.sort(axis=1), expected.sort(axis=1))
```

Comment:
the column order here should be based on the order of the fields in the dshape. I do not think that we should sort the columns of the `result` dataframe because it could mask bugs where the columns are not in the correct order. we could also be explicit about this with `expected` by using: `pd.DataFrame([['a', 2]], columns=['A', 'b'])`